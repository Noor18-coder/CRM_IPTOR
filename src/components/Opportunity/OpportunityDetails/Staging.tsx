import React from 'react'

const Staging = () => {
    return (
        // <!-- STAGING SECTION START -->
      <section className="sec-staging">
        <div className="d-flex justify-content-between title-row">
          <div className="lft-col">
            Stage
          </div>
          <div className="rgt-col">
            Shared for approval with Lawerence Matthew <a className="ghost-btn" href="#">Change Approver</a>
          </div>
        </div>

        <div className="staging-highlight">

          <ul className="hbreadcrumb">
            <li><a href="#">A0 <span>Initial</span></a></li>
            <li><a href="#">A1 <span>Initiated</span></a></li>
            <li><a href="#">A2 <span>Evaluate</span></a></li>
            <li><a href="#" className="active">A3 <span>Validate</span></a></li>
            <li><a href="#" className="ready">A4 <span>Demonstrate</span></a></li>
            <li><a href="#" className="ready">A5 <span>Negotiate</span></a></li>
            <li><a href="#">A6 <span>Closed</span></a></li>
          </ul>
        </div>
      </section>
    //   <!-- STAGING SECTION END -->
    )
}

export default Staging
