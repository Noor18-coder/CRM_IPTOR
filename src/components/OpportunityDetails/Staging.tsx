import React from 'react';
import { Dispatch } from 'redux';
import { useSelector, useDispatch } from 'react-redux';
import { StageInfo, AddOpportunityDefaultParams } from '../../helpers/Api/models';
import { saveOpportunityStages } from '../../store/InitialConfiguration/Actions';
import { editOpportunity, openOpportunityForm } from '../../store/OpportunityDetails/Actions';
import { AppState } from '../../store/store';
import { APPROVAL_STATUS } from '../../config/Constants';

// interface Props {
//   stage: string;
//   status: any;
// }

const Staging: React.FC = () => {
  const state: AppState = useSelector((appState: AppState) => appState);
  const { approvalStatus, stage } = state.opportuntyDetails.opportunityDefaultParams;
  const dispatch: Dispatch<any> = useDispatch();

  const updateStage = async (obj: StageInfo) => {
    const opportunity = { ...state.opportuntyDetails.opportunityDefaultParams };
    opportunity.stage = obj.salesStage;
    dispatch(editOpportunity(opportunity));
  };

  React.useEffect(() => {
    if (!state.enviornmentConfigs.crmOpportunityStage.length) dispatch(saveOpportunityStages());
  }, []);

  const changeActiveStatus = (e: React.ChangeEvent<HTMLInputElement>) => {
    const active: boolean = e.currentTarget.checked;
    if (active) {
      dispatch(openOpportunityForm({ open: true, groupName: 'deactivate-opportunity', action: 'edit', activ: active }));
    } else {
      const opportunityDetails: AddOpportunityDefaultParams = { ...state.opportuntyDetails.opportunityDefaultParams };
      opportunityDetails.activ = true;
      dispatch(editOpportunity(opportunityDetails));
    }
  };

  return (
    <>
      <div className="staging-highlight d-flex justify-content-between title-row">
        <div className="lft-col stages">
          <ul className="hbreadcrumb stage-lists">
            {state.enviornmentConfigs.crmOpportunityStage.map((obj: StageInfo) => {
              if (stage === obj.salesStage) {
                return (
                  <li
                    className={
                      approvalStatus === APPROVAL_STATUS.NEW
                        ? 'list-inline-item active'
                        : approvalStatus === APPROVAL_STATUS.REJECTED
                        ? 'list-inline-item rejected'
                        : approvalStatus === APPROVAL_STATUS.SUBMITTED
                        ? 'list-inline-item submit'
                        : approvalStatus === APPROVAL_STATUS.APPROVED
                        ? 'list-inline-item active'
                        : approvalStatus === APPROVAL_STATUS.WON
                        ? 'list-inline-item active'
                        : approvalStatus === APPROVAL_STATUS.LOST
                        ? 'list-inline-item lost'
                        : 'list-inline-item normal'
                    }>
                    {obj.salesStage} <span>{obj.description}</span>
                  </li>
                );
              } else {
                // eslint-disable-next-line no-lonely-if
                if (state.opportuntyDetails.editOportunity.allowEdit) {
                  return (
                    <li className="list-inline-item normal" role="presentation" onClick={() => updateStage(obj)}>
                      {obj.salesStage} <span>{obj.description}</span>
                    </li>
                  );
                } else {
                  return (
                    <li className="list-inline-item normal disabled" role="presentation">
                      {obj.salesStage} <span>{obj.description}</span>
                    </li>
                  );
                }
              }
            })}
          </ul>
        </div>
        {state.opportuntyDetails.editOportunity.allowEdit ? (
          <div className="rgt-col">
            <p className="oppt-lost">Opportunity Lost</p>
            <label className="switch">
              <input
                type="checkbox"
                id="oppty-lost"
                tabIndex={0}
                checked={!state.opportuntyDetails.opportunityDefaultParams.activ}
                onChange={changeActiveStatus}
              />
              <span className="slider round" />
            </label>
          </div>
        ) : null}
      </div>
    </>
  );
};

export default Staging;
