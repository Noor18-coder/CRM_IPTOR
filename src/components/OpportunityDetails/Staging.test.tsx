import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import Staging from './Staging';
import StoreMock from '../../mocks/Store.mock';

const store = StoreMock.createInitialStore();

describe('[OpportunityDetails] ProductAccordian', () => {
  it('should renders correctly', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <Staging />
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
