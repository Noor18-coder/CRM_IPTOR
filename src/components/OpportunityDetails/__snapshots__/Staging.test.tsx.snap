// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[OpportunityDetails] ProductAccordian should renders correctly 1`] = `
<div
  className="staging-highlight d-flex justify-content-between title-row"
>
  <div
    className="lft-col stages"
  >
    <ul
      className="hbreadcrumb stage-lists"
    />
  </div>
</div>
`;
